include/zephyr/modem/ubx.h

```diff
@@ -53,7 +53,7 @@ struct modem_ubx_match {
 
 struct modem_ubx_script {
 	struct {
-		const struct ubx_frame *buf;
+		const uint8_t *buf;
```

Comment:
if you do this then drop the casting in `subsys/modem/modem_ubx.c` `modem_ubx_run_script` since at that point it's not needed anymore,that said, since there's no point in the upstream code that accesses the data here as a byte array another option would be to change this to a `const void *` instead, that way you'd be able to drop all the castings in this PR and only keep the one in `modem_ubx_run_script`, and I don't think you'd lose anything really, the compiler would only make sure that the data pointed is constant and that's all you want.